apps/site/lib/site_web/plugs/canonical_hostname.ex

```diff
@@ -27,8 +27,12 @@ defmodule SiteWeb.Plugs.CanonicalHostname do
       conn
     else
       rewritten_url =
-        Plug.Conn.request_url(conn)
-        |> String.replace(requested_hostname, canonical_hostname, global: false)
+        if requested_hostname == "www.mbtace.com" do
+          "https://www.mbta.com" <> conn.request_path
+        else
+          Plug.Conn.request_url(conn)
+          |> String.replace(requested_hostname, canonical_hostname, global: false)
+        end
```

Comment:
The [code for request_url/1](https://github.com/elixir-plug/plug/blob/v1.12.1/lib/plug/conn.ex#L1676-L1685) looks like it shouldb e taking the scheme of the request into account. If it thinks that the scheme is `http`, perhaps the `Plug.SSL` plug is mis-configured?

Reply:
Oh, I see: this plug is configured in `SiteWeb.Endpoint` before `SiteWeb.Router`, so it runs before`Plug.SSL` has a chance to correct the scheme.

Reply:
Oh my gosh thank you!!! I was trying to figure out for the longest time why `Plug.SSL` wasn't helping us out here, hmmmm!

Reply:
Can you push up a branch with what you have? I'll take a look tomorrow.

Reply:
@paulswartz I eventually realized I can add some stuff to the test config to make the test behave more in line with the prod environment, so I think I've got it now. Pushed up the changes.